Polyhedron/demo/Polyhedron/include/CGAL/IO/read_surf_trianglemesh.h

```diff
@@ -0,0 +1,186 @@
+#include <iostream>
+#include <string>
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_function_params.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_params_helper.h>
+
+//a material is composed of an material Id and a material name.
+typedef std::pair<int, std::string> material;
+
+struct MaterialData
+{
+  material innerRegion;
+  material outerRegion;
+};
+
+/*!
+ * read_surf reads a file which extension is .surf and fills `output` with one `Mesh` per patch.
+ * Mesh is a model of FaceListGraph.
+ */
+template<class Mesh, class NamedParameters>
+void read_surf(std::istream& input, std::vector<Mesh>& output, std::vector<MaterialData>& metadata,  const NamedParameters&)
+{
+  typedef typename CGAL::GetGeomTraits<Mesh,
+      NamedParameters>::type Kernel;
+  typedef typename Kernel::Point_3 Point_3;
+  typedef typename boost::graph_traits<Mesh>::vertex_descriptor vertex_descriptor;
+  std::vector<Point_3> points;
+  std::string line;
+  std::istringstream iss;
+  std::size_t nb_vertices(0);
+  std::vector<material> materials;
+  //ignore header
+  while(std::getline(input, line))
+    if(line.compare(0, 13, "    Materials") != 0)
```

Comment:
this (and all similar calls to `line.compare()` below is not robust to indentation changes.
You can use [std::string::find_first_not_of](http://www.cplusplus.com/reference/string/string/find_first_not_of/) or [std::isspace](http://en.cppreference.com/w/cpp/string/byte/isspace) and `std::string::erase` to remove spaces at the beginning of the line and make it more robust.